Check for internet connection

ansraliant opened the pull request:
Hello, first of all let me thank you for your script, it's very useful and I use it every day on my many open suse installations. 

I made this Pull Request out of a little annoyance I got the other day. I tried to update but forgot to check my internet connection, and the script tried to update anyway, nothing happened, but I thought it would be nice a little message. So this is what this PR is about, just a simple check for the version obtained with curl, if it's empty, because something happen display a message and exit, nothing more.

Let me know what you think.

Thank you.

ansraliant committed:
Check for internet connection

When doing an upgrade, before letting the user continue with the
operation, if the version obtained from curl is empty (something
happened to curl that could not get the version) return a simple message
to the user.

boombatower commented on tumbleweed:
```diff
@@ -161,6 +161,11 @@ tumbleweed_switch()
     echo "choosing latest version"
   fi
 
+  if [ -z "$version" ] ; then
```
I believe the preferred way to check empty variable is `[ -z ${version+x}` ]`.

boombatower commented on tumbleweed:
```diff
@@ -161,6 +161,11 @@ tumbleweed_switch()
     echo "choosing latest version"
   fi
 
+  if [ -z "$version" ] ; then
```
I believe the preferred way to check empty variable is `[ -z ${version+x}` ]`. Perhaps move this into the `tumbleweed_latest` part of conditional local since the other already performs a check.

ansraliant replied:
I tried with `[ -z ${version+x} ]` and it didn't catch it. But if I do it `[[ -z $version ]]` it works.

boombatower commented on tumbleweed:
```diff
@@ -161,6 +161,11 @@ tumbleweed_switch()
   fi
 
+  if [ -z "$version" ] ; then
+    echo "There is no version to upgrade to. Please check your internet connection"
```
Perhaps, `Unable to determine latest version (likely due to network connection issue)` as `There is no version to upgrade to` is not necessarily true, but the script cannot determine if that is the case.

ansraliant replied:
That is true, that message is more concise.

boombatower reviewed (changes requested):
Small changes if you would, thanks for the submission.

ansraliant committed:
Moved Condition and change message

The validation of the version has been moved and
the message regarding the network connection has been changed

ansraliant commented:
Sorry for the delay. I have applied your comments. Let me know what you think.
I misunderstood a comment, that's why I amended my commit.
Thank you.

boombatower commented:
Tweaked slightly after adding `--fail` to all curl calls. Thanks for the contribution and feedback.

boombatower commented:
Updated package and [submitted to Factory](https://build.opensuse.org/request/show/682649).
